homeassistant/components/forked_daapd/media_player.py

```diff
@@ -0,0 +1,476 @@
+"""This library brings support for forked_daapd to Home Assistant."""
+import asyncio
+from collections import defaultdict
+from functools import partialmethod
+import logging
+
+from pyforked_daapd import ForkedDaapdAPI, ForkedDaapdData
+import voluptuous as vol
+
+from homeassistant.components.media_player import PLATFORM_SCHEMA, MediaPlayerDevice
+from homeassistant.components.media_player.const import (
+    SUPPORT_CLEAR_PLAYLIST,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SEEK,
+    SUPPORT_SHUFFLE_SET,
+    SUPPORT_STOP,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_PAUSED,
+    STATE_PLAYING,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.dt import utcnow
+
+_LOGGER = logging.getLogger(__name__)
+SUPPORTED_FEATURES = (
+    SUPPORT_PLAY
+    | SUPPORT_PAUSE
+    | SUPPORT_STOP
+    | SUPPORT_SEEK
+    | SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_MUTE
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_CLEAR_PLAYLIST
+    | SUPPORT_SHUFFLE_SET
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+)
+
+SUPPORTED_FEATURES_ZONE = (
+    SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+)
+
+DOMAIN = "forked_daapd"
+DEFAULT_PORT = 3689
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler